components/Forms/AddDeviceForm/AddDeviceForm.js

```diff
@@ -71,68 +156,88 @@ export default function AddDeviceForm() {
   };
 
   return (
-    <StyledFormContainer onSubmit={handleSubmit(onSubmit)}>
-      <input
-        type="checkbox"
-        value="manual"
-        {...register("generateData")}
-        onChange={handleToggle}
-        checked={isAutomatic}
-      />
-      <StyledLabel>
-        <StyledInput
-          {...register("name", {
-            required: true,
-            maxLength: 20,
-          })}
-        />
-        {errors.name && <StyledError>Please type in a Device Name</StyledError>}
-      </StyledLabel>
-      <StyledLabel>
-        <StyledInput
-          {...register("location", { required: true, maxLength: 20 })}
-        />
-        {errors.location && (
-          <StyledError>Pleace type in a Location</StyledError>
-        )}
-      </StyledLabel>
-      <StyledLabel>
-        <select {...register("type", { required: true })}>
-          <option value="Please Select Device" disabled selected>
-            Please Select Device
-          </option>
-          <option value="Refrigerator">Refrigerator</option>
-          <option value="Freezer">Freezer</option>
-          <option value="Prep Table">Prep Table</option>
-        </select>
-        {errors.type && <StyledError>Please select a Devicetype</StyledError>}
-      </StyledLabel>
-      <StyledLabel>
-        <StyledInput
-          type="number"
-          placeholder="min Temp"
-          {...register("minTemp", { required: true, min: -25, max: 25 })}
-        />
-        {errors.minTemp && (
-          <StyledError>
-            Please enter a temperature between -25°C and 25°C
-          </StyledError>
-        )}
-      </StyledLabel>
-      <StyledLabel>
-        <StyledInput
-          type="number"
-          placeholder="max Temp"
-          {...register("maxTemp", { required: true, min: -25, max: 25 })}
-        />
-        {errors.maxTemp && (
-          <StyledError>
-            Please enter a temperature between -25°C and 25°C
-          </StyledError>
-        )}
-      </StyledLabel>
-      <button type="submit">Submit</button>
-      {submitMessage && <StyledSubmit>Device Succesfully Added!</StyledSubmit>}
-    </StyledFormContainer>
+    <>
+      <HeaderNavigation />
+      <StyledWrapper>
+        <StyledFormContainer onSubmit={handleSubmit(onSubmit)}>
```

Comment:
what exactly does `handleSubmit` do here? If you just want to invoke `onSubmit`, you could use `() => onSubmit` instead.

Reply:
handleSubmit is a Callback Function from the react-form-hook, it receives an object containing the form data as its argument when it's submitted, so I used that because of the hook.